packages/walletconnect-v2/src/index.ts

```diff
@@ -48,27 +61,30 @@ export class WalletConnect extends Connector {
   public provider?: WalletConnectProvider
   public readonly events = new EventEmitter3()
 
-  private readonly options: Omit<WalletConnectOptions, 'rpcMap' | 'chains'>
-
-  private readonly rpcMap?: Record<number, string | string[]>
-  private readonly chains: number[]
   private readonly defaultChainId?: number
-  private readonly timeout: number
-
   private eagerConnection?: Promise<WalletConnectProvider>
+  private readonly options: Omit<WalletConnectOptions, 'rpcMap'>
+  private readonly rpcMap?: Record<number, string | string[]>
+  private readonly timeout: number
 
-  constructor({ actions, options, defaultChainId, timeout = DEFAULT_TIMEOUT, onError }: WalletConnectConstructorArgs) {
+  constructor({ actions, defaultChainId, options, timeout = DEFAULT_TIMEOUT, onError }: WalletConnectConstructorArgs) {
     super(actions, onError)
 
-    const { rpcMap, rpc, chains, ...rest } = options
-
-    this.options = rest
-    this.chains = chains
```

Comment:
IMO this should not be removed. The principle was that all the props that are custom modified / not passed straight to WalletConnect should be kept outside to avoid confusion.

Reply:
I was taking the opposite approach. If this is intended, we should at least include a comment explaining that we're intentionally *not* trying to be a thin pass-through layer.

Reply:
I think this is exactly what we're trying to do - to be a thin pass-through layer. It's just that internally we have made this decision to differentiate between properties that we transform vs the ones we pass-through to make it easier to maintain.

That way, you can e.g. check for arrays during initialisation time and put them on class as a property for further reference.